src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -1548,6 +1549,45 @@ public C(string s)
 }}", index: 2);
         }
 
+        [WorkItem(51338, "https://github.com/dotnet/roslyn/issues/51338")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestSpecialStringCheck3()
+        {
+            var culture = CultureInfo.CurrentUICulture;
+            try
+            {
+                CultureInfo.CurrentUICulture = new CultureInfo("de-DE", useUserOverride: false);
+
+                await TestInRegularAndScript1Async(
+    @"
+using System;
+
+class C
+{
+    public C([||]string s)
+    {
+    }
+}",
+    @$"
+using System;
+
+class C
+{{
+    public C(string s)
+    {{
+        if (string.IsNullOrEmpty(s))
+        {{
+            throw new ArgumentException($""{string.Format(FeaturesResources._0_cannot_be_null_or_empty, "{nameof(s)}").Replace("\"", "\\\"")}"", nameof(s));
```

Comment:
If, for any weird reason, this resource for this specific language was updated in the future to use single quotes, the test will not be adding anything. Consider one of the following:

- Add an assertion to make sure this resource containing `"{0}"`
- Copy the exact resource message into the test

Reply:
I did think about doing something along those lines, but having a test fail during a LOC insertion is not something I really want to do, especially when neither of those failure cases indicates a regression of the fix.

Reply:
> having a test fail during a LOC insertion is not something I really want to do, especially when neither of those failure cases indicates a regression of the fix.

Sounds reasonable